precompiles/outposts/osmosis/tx_test.go

```diff
@@ -0,0 +1,342 @@
+					transferAmount,
+					slippagePercentage,
+					windowSeconds,
+					osmoAddress,
+				}
+			},
+			expError:    true,
+			errContains: fmt.Sprintf("token '%s' not registered", randomAddress),
+		}, {
+			name:   "fail - missing output token denom",
+			sender: senderAddress,
+			origin: senderAddress,
+			malleate: func() []interface{} {
+				evmosTokenPair, err := testutils.RegisterEvmosERC20Coins(*s.unitNetwork, sender)
+				s.Require().NoError(err, "expected no error during evmos erc20 registration")
+
+				return []interface{}{
+					senderAddress,
+					evmosTokenPair.GetERC20Contract(),
+					randomAddress,
+					transferAmount,
+					slippagePercentage,
+					windowSeconds,
+					osmoAddress,
+				}
+			},
+			expError:    true,
+			errContains: fmt.Sprintf("token '%s' not registered", randomAddress),
+		}, {
+			name:   "fail - osmo token pair not registered (with osmo hardcoded address)",
+			sender: senderAddress,
+			origin: senderAddress,
+			malleate: func() []interface{} {
+				evmosTokenPair, err := testutils.RegisterEvmosERC20Coins(*s.unitNetwork, sender)
+				s.Require().NoError(err, "expected no error during evmos erc20 registration")
+
+				return []interface{}{
+					senderAddress,
+					common.HexToAddress("0x1D54EcB8583Ca25895c512A8308389fFD581F9c9"),
+					evmosTokenPair.GetERC20Contract(),
+					transferAmount,
+					slippagePercentage,
+					windowSeconds,
+					osmoAddress,
+				}
+			},
+			expError:    true,
+			errContains: fmt.Sprintf("token '%s' not registered", common.HexToAddress("0x1D54EcB8583Ca25895c512A8308389fFD581F9c9")),
+		}, {
+			name:   "fail - osmo token pair registered with another channelID",
+			sender: senderAddress,
+			origin: senderAddress,
+			malleate: func() []interface{} {
+				evmosTokenPair, err := testutils.RegisterEvmosERC20Coins(*s.unitNetwork, sender)
+				s.Require().NoError(err, "expected no error during evmos erc20 registration")
+
+				osmoIbcDenomTrace := utils.ComputeIBCDenomTrace(portID, channelID, osmosis.OsmosisDenom)
+				_, err = testutils.RegisterIBCERC20Coins(*s.unitNetwork, sender, osmoIbcDenomTrace)
+				s.Require().NoError(err, "expected no error during ibc erc20 registration")
+
+				wrongOsmoIbcDenomTrace := utils.ComputeIBCDenomTrace(portID, "channel-1", osmosis.OsmosisDenom)
+				wrongOsmoTokenPair, err := testutils.RegisterIBCERC20Coins(*s.unitNetwork, sender, wrongOsmoIbcDenomTrace)
+				s.Require().NoError(err, "expected no error during ibc erc20 registration")
+
+				return []interface{}{
+					senderAddress,
+					wrongOsmoTokenPair.GetERC20Contract(),
+					evmosTokenPair.GetERC20Contract(),
+					transferAmount,
+					slippagePercentage,
+					windowSeconds,
+					osmoAddress,
+				}
+			},
+			expError: true,
+			// Probably there is a better way than hardcoding the expected string
+			errContains: fmt.Sprintf(osmosis.ErrInputTokenNotSupported, []string{"aevmos", "ibc/ED07A3391A112B175915CD8FAF43A2DA8E4790EDE12566649D0C2F97716B8518"}),
```

Comment:
nit: you can define this expected error at the top of the file with a name and add it here afterwards.

Reply:
Isn't better to maintain consistency in the code? Because moving only this one on the top will end up having this errContains field built in a different way than others.